pkg/controllers/datastoresyncer/datastoresyncer.go

```diff
@@ -74,33 +74,35 @@ func NewDatastoreSyncer(thisClusterID string, objectNamespace string, kubeClient
 	return &newDatastoreSyncer
 }
 
-func (d *DatastoreSyncer) ensureExclusiveEndpoint() {
+func (d *DatastoreSyncer) ensureExclusiveEndpoint() error {
 	klog.V(4).Infof("Ensuring we are the only endpoint active for this cluster")
 	endpoints, err := d.datastore.GetEndpoints(d.localCluster.ID)
 	if err != nil {
-		klog.Fatalf("Error while retrieving endpoints %v", err)
+		return fmt.Errorf("error retrieving endpoints %v", err)
 	}
 
 	for _, endpoint := range endpoints {
 		if !util.CompareEndpointSpec(endpoint.Spec, d.localEndpoint.Spec) {
 			endpointCrdName, err := util.GetEndpointCRDName(endpoint)
 			if err != nil {
-				klog.Errorf("Error while converting endpoint to CRD Name %s", endpoint.Spec.CableName)
-				break
+				klog.Errorf("error converting endpoint %#v to CRD Name: %v", endpoint, err)
```

Comment:
What’s the reasoning here (and elsewhere in the loop) which leads to logging an error rather than returning the error?

Reply:
Originally I changed these sites to return error as well but I'm unsure as to the original intent. These are questions went thru my mind: Why was the error above fatal but not this one and the 2 below it... Why does it break here and not below...  Are these errors catastrophic enough to propagate and thus exit the process...  When deleting, would it return an error if the object doesn't exist...    The latter 2 made me squeamish about changing the behavior.

Reply:
@Oats87 what do you think?